deeppavlov/models/multitask_pal_bert/multitask_pal_bert.py

```diff
@@ -0,0 +1,468 @@
+            task_features = in_by_tasks[self.task_names[task_id]]
+
+            _input = {}
+            for elem in ["input_ids", "attention_mask", "token_type_ids"]:
+                _input[elem] = [getattr(f, elem) for f in task_features[0]]
+                _input[elem] = torch.cat(_input[elem], dim=0).to(self.device)
+
+            with torch.no_grad():
+                tokenized = {
+                    key: value
+                    for (key, value) in _input.items()
+                    if key in self.model.forward.__code__.co_varnames
+                }
+
+                logits = self.model(
+                    task_id=task_id, name=self.tasks_type[task_id], **tokenized
+                )
+
+            if self.return_probas:
+                if not self.multilabel:
+                    pred = torch.nn.functional.softmax(logits, dim=-1)
+                else:
+                    pred = torch.nn.functional.sigmoid(logits)
+                pred = pred.detach().cpu().numpy()
+            elif self.tasks_num_classes[task_id] > 1:
+                logits = logits.detach().cpu().numpy()
+                pred = np.argmax(logits, axis=1)
+            else:  # regression
+                pred = logits.squeeze(-1).detach().cpu().numpy()
+            self.validation_predictions.append(pred)
+        return self.validation_predictions
+
+    def train_on_batch(self, *args):
+        """Train model on given batch.
+        This method calls train_op using features and y (labels).
+        Args:
+            features: batch of InputFeatures
+            y: batch of labels (class id)
+        Returns:
+            dict with loss for each task
+        """
+        n_in = sum([inp for inp in self.in_distribution.values()])
+        task_id = args[0]
+        features = args[1:]
+        args_in, args_in_y = features[:n_in], features[n_in:]
+        in_by_tasks = self._distribute_arguments_by_tasks(
+            args_in, {}, self.task_names, "in"
+        )
+        in_y_by_tasks = self._distribute_arguments_by_tasks(
+            args_in_y, {}, self.task_names, "in_y"
+        )
+
+        task_features = in_by_tasks[self.task_names[task_id]]
+        task_labels = in_y_by_tasks[self.task_names[task_id]]
+
+        _input = {}
+        for elem in ["input_ids", "attention_mask", "token_type_ids"]:
+            _input[elem] = [getattr(f, elem) for f in task_features[0]]
+            _input[elem] = torch.cat(_input[elem], dim=0).to(self.device)
+
+        if self.tasks_type[task_id] == "regression":
+            _input["labels"] = torch.tensor(
+                np.array(task_labels[0], dtype=float), dtype=torch.float32
+            ).to(self.device)
+        else:
+            _input["labels"] = torch.from_numpy(np.array(task_labels[0])).to(
+                self.device
+            )
+        tokenized = {
+            key: value
+            for (key, value) in _input.items()
+            if key in self.model.forward.__code__.co_varnames
```

Comment:
Could you explain please if we really need this condition, and if yes, why?

Reply:
Removed as it is not required.